pennylane/circuit_drawer/representation_resolver.py

```diff
@@ -0,0 +1,406 @@
+        """Format an operation that corresponds to a single matrix.
+
+        Args:
+            operation (~.Operation): Operation that shall be formatted
+            symbol (str): The symbol that should be used to identify matrices
+            cache (List[numpy.ndarray]): The cache of already known matrices
+
+        Returns:
+            str: The formatted operation
+        """
+        mat = operation.params[0]
+        idx = RepresentationResolver.index_of_array_or_append(mat, cache)
+
+        return "{}{}".format(symbol, idx)
+
+    @staticmethod
+    def _format_matrix_arguments(params, symbol, cache):
+        """Format a sequence of matrix parameters.
+
+        Args:
+            params (List[numpy.ndarray]): List of matrix parameters
+            symbol (str): The symbol that should be used to identify matrices
+            cache (List[numpy.ndarray]): The cache of already known matrices
+
+        Returns:
+            str: The formatted matrix arguments
+        """
+        param_strings = []
+        for param in params:
+            idx = RepresentationResolver.index_of_array_or_append(param, cache)
+
+            param_strings.append("{}{}".format(symbol, idx))
+
+        return "(" + ",".join(param_strings) + ")"
+
+    @staticmethod
+    def _format_poly_term(coefficient, variable):
+        """Format a term in a polynomial.
+
+        Args:
+            coefficient (float): The polynomial term's coefficient
+            variable (str): The polynomial term's variable
+
+        Returns:
+            str: The formatted term
+        """
+        if coefficient == 0:
+            return ""
+
+        if coefficient == 1.0:
+            return str(variable)
+
+        if coefficient == -1.0:
+            return "-" + str(variable)
+
+        return "{:+.3g}{}".format(coefficient, variable)
+
+    def _format_polyxp(self, operation):
+        """Format a polynomial of x and p operators.
+
+        Theses operators appear as observables in CV quantum computing.
+
+        Args:
+            operation (~.PolyXP): The PolyXP observable that shall be formatted.
+
+        Returns:
+            str: A string representing the polynomial
+        """
+        coefficients = operation.params[0]
+        order = len(coefficients.shape)
+
+        if order == 1:
+            poly_str = ""
+            if coefficients[0] != 0:
+                poly_str += "{:.3g}".format(coefficients[0])
+
+            for idx in range(0, coefficients.shape[0] // 2):
+                x = 2 * idx + 1
+                y = 2 * idx + 2
+                poly_str += RepresentationResolver._format_poly_term(
+                    coefficients[x], "x{}".format(self.charset.to_subscript(idx))
+                )
+                poly_str += RepresentationResolver._format_poly_term(
+                    coefficients[y], "p{}".format(self.charset.to_subscript(idx))
+                )
+
+            return poly_str
```

Comment:
Looks really nice that it has its own method! :blush: 

Another step could even be breaking it up into an ``order==1`` and ``order==2`` cases, if I see it correctly they would each require passing only the ``coefficients`` as an argument.

This way this bigger method would also be broken down.

Reply:
That's a good suggestion, I implemented it!